### PR TITLE
executor: tiny refactor, remove `physicalTableID` from the three reader

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Tiny refactor

### What is changed and how it works?

After https://github.com/pingcap/tidb/pull/8871, the `e.table` in the TableReader, IndexReader, IndexLookupReader are actually `PhysicalTable`, so the physicalTableID is redundant, it can be extracted from the `e.table`.

### Check List 

Tests 

 - Unit test
 - No code

### CODE REVIEW

File: executor/distsql.go [hunk 1]
```diff
@@ -213,6 +213,7 @@ func rebuildIndexRanges(ctx sessionctx.Context, is *plannercore.PhysicalIndexSca
 type IndexReaderExecutor struct {
 	baseExecutor
 
+	// Table is a PhysicalTable for partitioned table.
```
Comment (eurekaka) [inline executor/distsql.go]: This comment looks weird, can we elaborate it more or just remove it?

Comment (tiancaiamao) [reply executor/distsql.go]: Updated, is it better now?

**Pull request merged**